docs/src/userguide/source_files_and_compilation.rst

```diff
@@ -833,6 +833,14 @@ Cython code.  Here is the list of currently supported directives:
     appropriate exception is raised. This is off by default for
     performance reasons.  Default is False.
 
+``freethreading_compatible``  (True / False)
+    If set to True, Cython sets the ``Py_mod_gil`` slot to
+    ``Py_MOD_GIL_NOT_USED`` to signal that the module is safe to run
+    without an active GIL and prevent the GIL from being enabled
+    when the module is imported. Otherwise the slot is set to
+    ``Py_MOD_GIL_USED`` which will cause the GIL to be automatically
```

Comment:
I'd suggest adding something like:

"Setting this to True does not itself make the module safe to run without the GIL; it merely confirms that you have checked the logic and consider it safe to run."

Just so people don't believe that it does more than set the flag.